### PR TITLE
specification for time aware issuers/keys

### PR DESCRIPTION
**Pull request by husobee**
https://github.com/brave-intl/challenge-bypass-server/issues/106 attempted to flesh this out

### CODE REVIEW

File: kafka/signed_blinded_token_issuer_handler.go [hunk 1]
```diff
@@ -75,61 +87,107 @@ func SignedBlindedTokenIssuerHandler(
...
+					logger.Error().Msg(fmt.Sprintf("Request %s: failed to update associated data: %e", blindedTokenRequestSet.Request_id, err))
+					blindedTokenResults = append(blindedTokenResults, avroSchema.SigningResult{
+						Signed_tokens:     nil,
+						Issuer_public_key: "",
+						Status:            ERROR,
+						Associated_data:   request.Associated_data,
+					})
+					continue
+				}
+				// add valid from / valid to to the associated data
```
Comment (clD11) [inline kafka/signed_blinded_token_issuer_handler.go:128]: typo to to

File: kafka/signed_blinded_token_issuer_handler.go [hunk 2]
```diff
@@ -75,61 +87,107 @@ func SignedBlindedTokenIssuerHandler(
...
+				request.Associated_data = generated.Bytes(associatedDataJson)
```
Comment (clD11) [inline kafka/signed_blinded_token_issuer_handler.go:129]: Does this need converted already []byte

File: kafka/signed_blinded_token_issuer_handler.go [hunk 3]
```diff
@@ -56,6 +58,16 @@ func SignedBlindedTokenIssuerHandler(
 			})
 			continue
 		}
+		// validate that the number of blinded tokens is evenly divisable
+		if len(request.Blinded_tokens)%issuer.Buffer != 0 {
+			blindedTokenResults = append(blindedTokenResults, avroSchema.SigningResult{
+				Signed_tokens:     nil,
```
Comment (clD11) [inline kafka/signed_blinded_token_issuer_handler.go:64]: should this be camelCase?

File: kafka/signed_blinded_token_issuer_handler.go [hunk 4]
```diff
@@ -56,6 +58,16 @@ func SignedBlindedTokenIssuerHandler(
 			})
 			continue
 		}
+		// validate that the number of blinded tokens is evenly divisable
+		if len(request.Blinded_tokens)%issuer.Buffer != 0 {
```
Comment (clD11) [inline kafka/signed_blinded_token_issuer_handler.go:62]: should this be camelCase?

File: kafka/signed_blinded_token_issuer_handler.go (hunk 3 above)
Comment (husobee) [reply kafka/signed_blinded_token_issuer_handler.go:64]: avro generated type doesn't follow our linting style unfortunately

File: kafka/signed_blinded_token_issuer_handler.go (hunk 2 above)
Comment (husobee) [reply kafka/signed_blinded_token_issuer_handler.go:129]: yeah, has to be converted to the avro bytes

File: server/issuers.go [hunk 5]
```diff
@@ -159,6 +170,47 @@ func (c *Server) issuerGetAllHandler(w http.ResponseWriter, r *http.Request) *ha
...
+	}); err != nil {
+		log.Errorf("%s", err)
+		return &handlers.AppError{
+			Cause:   err,
+			Message: "Could not create new issuer",
+			Code:    500,
+		}
+	}
+
+	w.WriteHeader(http.StatusOK)
```
Comment (clD11) [inline server/issuers.go:210]: should this be created

File: server/issuers.go [hunk 6]
```diff
@@ -159,6 +170,47 @@ func (c *Server) issuerGetAllHandler(w http.ResponseWriter, r *http.Request) *ha
...
+		if req.ExpiresAt.Before(time.Now()) {
+			c.Logger.Error("Expiration time has past")
+			return &handlers.AppError{
+				Message: "Expiration time has past",
+				Code:    400,
+			}
+		}
+	}
+
+	if err := c.createV3Issuer(Issuer{
```
Comment (clD11) [inline server/issuers.go:194]: Should the overlap also be added here

Comment (clD11) [reply server/issuers.go:194]: ValidFrom is part of the issuerV3CreateRequest but not used here should it be added

**Review by clD11 — APPROVED** [review]

**Review by clD11 — APPROVED** [review]

**Review by clD11 — APPROVED** [review]

**Pull request merged**